sphinx/util/requests.py

```diff
@@ -41,18 +35,6 @@
                      'Mozilla/5.0 (X11; Linux x86_64; rv:25.0) Gecko/20100101 Firefox/25.0')]
 
 
-def is_ssl_error(exc: Exception) -> bool:
```

Comment:
To avoid the error of 3rd party extensions, we usually obey the deprecation policy: https://www.sphinx-doc.org/en/master/internals/release-process.html?highlight=deprecation#deprecation-policy.
So please don't remove this function, and add a deprecation warning to the top of the function instead.
46f0b3063d is an example of deprecation.

I know nobody accesses this function. So I agree this is a bit boring work. But Sphinx has many extensions. And some of them touches the inside of Sphinx to hack our software. So we need to change the interface of class and functions carefully.

Reply:
Thanks for the explanation and pointers, it was definitely helpful! I’ll keep the deprecation policy in mind for future patches.

I restored `is_ssl_error()` and added a `RemovedInSphinx50Warning`. I believe the warning message make clear that the entire function should be removed?

Reply:
Looks good. Additionally, please add an entry to CHANGES and docs/extdev/deprecated.rst. These lists are also useful for developers.

Reply:
Added an entry to both documents. Thanks for the review!